source/comp/markv.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// MARK-V is a compression format for SPIR-V binaries. It strips away
+// non-essential information (such as result ids which can be regenerated) and
+// uses various bit reduction techiniques to reduce the size of the binary and
+// make it more similar to other compressed SPIR-V files to further improve
+// compression of the dataset.
+
+#ifndef SPIRV_TOOLS_MARKV_HPP_
+#define SPIRV_TOOLS_MARKV_HPP_
+
+#include <string>
+#include <vector>
+
+#include "markv_model.h"
```

Comment:
order of includes.  headers from outside this module should precede headers from inside this module.

Reply:
https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes says

-----------
In dir/foo.cc or dir/foo_test.cc, whose main purpose is to implement or test the stuff in dir2/foo2.h, order your includes as follows:

dir2/foo2.h.
C system files.
C++ system files.
Other libraries' .h files.
Your project's .h files
----------

So it depends on your interpretation of "module".  The rule uses "library" and "project" in a possibly ambiguous way.  I take "module" = "library" and the markv_model.h is in the SPIRV-Tools-comp library, while libspriv.hpp declares content from other SPIRV-Tools libaries.

Do you agree?

Reply:
Perhaps this is too small to argue over.  The rest of SPIRV-Tools is not uniformly good in the header include order department.